lib/apiservers/engine/backends/container.go

```diff
@@ -667,6 +667,11 @@ func (c *ContainerBackend) ContainerCreate(config types.ContainerCreateConfig) (
 
 	var err error
 
+	if err = checkContainerCount(op, PortLayerClient()); err != nil {
+		return containertypes.ContainerCreateCreatedBody{}, err
+	}
+	defer cache.ContainerCache().DecreaseContainerCount()
```

Comment:
What's the logic of this code?

Reply:
checkContainerCount will reserve one container request by increase reservation number. When container creation finishes, we need to decrease the reservation number, as the container creation has finished.